ui/src/logic/usePathFromHark.ts

```diff
@@ -0,0 +1,46 @@
+import { udToDec } from '@urbit/api';
+import { useWritByFlagAndWritId } from '@/state/chat';
+import { Skein } from '@/types/hark';
+
+export default function usePathFromHark(bin: Skein, isMention: boolean) {
+  const writId = bin.top?.wer.split('/')[10];
+  const ship =
+    typeof bin.top.con[0] !== 'string' && 'ship' in bin.top.con[0]
+      ? bin.top?.con[0].ship
+      : '';
+  const channelFlag = `${bin.top?.wer.split('/')[6]}/${
+    bin.top?.wer.split('/')[7]
+  }`;
+  const idWrit = `${ship}/${writId}`;
+  const replyToWrit = useWritByFlagAndWritId(channelFlag, idWrit || '', false)
+    ?.memo.replying;
+  const wer = bin.top?.wer;
+
+  if (!isMention) {
+    return wer;
+  }
```

Comment:
I think this needs to be at the top otherwise this will error out when trying to reference specific segments like `[10]`